app/bundles/EmailBundle/Helper/MailHelper.php

```diff
@@ -1505,13 +1505,7 @@ public function getCustomHeaders()
      */
     private function getUnsubscribeHeader()
     {
-        if ($this->idHash) {
-            $url = $this->factory->getRouter()->generate('mautic_email_unsubscribe', ['idHash' => $this->idHash], UrlGeneratorInterface::ABSOLUTE_URL);
-
-            return "<$url>";
-        }
-
-        if (!empty($this->queuedRecipients) || !empty($this->lead)) {
+        if (!empty($this->queuedRecipients) || !empty($this->lead) || $this->idHash) {
```

Comment:
I think this line needs review, original the two cases were separated based on the existence of ```idHash``` where we get the value of the unsubscribe link based on the hash, where in the queued ones we use the token.

I think the old file has the correct behavior and we do not need to change it.

Did you find a case where we need to use the change you propose?

Reply:
The problem with that is that idhash stay the same for all batch when its not queue so without this change issue will remains. If we want to keep using this we need to find a way to exclude or update idhash for the not queue model.

Reply:
Thanks @adirbnology I am doing testing using this PR

Reply:
This is not a correct fix. hashId must be unique for each record in the email_stat table. If it is not then the problem is in the AmazonApiTransport. Not here.

cc @alanhartless